go/worker/compute/executor/committee/node.go

```diff
@@ -258,23 +258,18 @@ func (n *Node) HandlePeerMessage(ctx context.Context, message *p2p.Message, isOw
 			return true, nil
 		}
 
-		// Check transaction before queuing it.
-		tx, err := n.checkTx(ctx, rawTx)
-		if err != nil {
-			return true, err
-		}
-
-		n.logger.Debug("queuing transaction",
-			"transaction", tx,
+		// Queue transaction for check tx .
+		n.logger.Debug("queuing transaction for check",
+			"tx", rawTx,
 		)
-		err = n.QueueTx(tx)
-		if err != nil {
+		if err := n.checkTxQueue.Add(rawTx); err != nil {
```

Comment:
One possible problem with this change is that nodes will now propagate bad transactions (e.g., ones that fail checks). Maybe this should block until the transaction in the batch passes the checks (or the context is cancelled) and propagate the correct error?

On the other hand this would delay transaction propagation until the whole batch is checked.

Reply:
> One possible problem with this change is that nodes will now propagate bad transactions

Hm yeah, but that I think that this is already the case even now, for all nodes other than the active primary executor workers (which is usually smaller than the backup committee)?